documentation/how-to/frontend-tls.md

```diff
@@ -0,0 +1,46 @@
+# Configure frontend TLS
+
+This guide provides instructions for configuring frontend TLS in Charmed Temporal.
+
+In the context of Temporal, frontend TLS refers to the use of TLS to secure network communication between Temporal clients (SDKs, CLI, or Web UI) and the Temporal Frontend Service. The Temporal Frontend Service acts as the API gateway for client requests to the Temporal Server.
+
+See [TLS configuration reference](https://docs.temporal.io/references/configuration#tls) for more details.
+
+[note]
+
+Frontend TLS is different from TLS termination at ingress. In the former case, TLS termination is handled by the Temporal frontend. If your setup requires TLS termination at ingress, see [Configure Ingress with Nginx Ingress Integrator](https://discourse.charmhub.io/t/charmed-temporal-k8s-tutorial-deploy-nginx-ingress-integrator/11783) guide.
+
+While it is technically possible to have end-to-end encryption with Temporal handling TLS termination and the ingress bypassing encrypted requests, this setup is not currently supported.
```

Comment:
what does `ingress bypassing encrypted requests` here mean? does it mean that ingress will proxy/forward encrypted requests? if so, what is preventing ingress from forwarding encrypted requests? also, if so, does it mean that if Frontend TLS termination occurs, the usage ingress is not supported?

Reply:
> what does ingress bypassing encrypted requests here mean?

It means that the ingress will transmit data to the frontend server without decrypting them, meaning the TLS termination will be done at the server and not at the ingress.

> does it mean that ingress will proxy/forward encrypted requests? if so, what is preventing ingress from forwarding encrypted requests?

encrypted data, rather, but yes. I should change the sentence -> `ingress bypassing encrypted data`

> also, if so, does it mean that if Frontend TLS termination occurs, the usage ingress is not supported?

Not at least for now. You'd have to go with either of these options.

Reply:
thanks for clarifying! i would then recommend the `ingress proxying encrypted data without tls termination` as a more precise statement. futhermore, please consider adding a note about either one or the other set up being supported, and that currently no support for both tls termination at ingress and temporal frotend

Reply:
Good suggestion, I have rewritten the paragraph to make this more explicit. Let me know what you think.